src/optimizer/plugins/CombineCodingAndQuantityValuesOptimizer.ts

```diff
@@ -137,3 +146,46 @@ export default {
     });
   }
 } as OptimizerPlugin;
+
+function moveUpCaretValueRule(
+  rule: ExportableCaretValueRule,
+  sd: ExportableProfile | ExportableExtension,
+  knownSiblings: ExportableCaretValueRule[]
+): void {
+  // if the caretPath ends with coding[0], and there are no sibling rules that use indices on coding,
+  // we can move the rule up a level to take advantage of how SUSHI handles rules on CodeableConcepts.
+  if (rule.caretPath.endsWith('coding[0]')) {
+    const basePath = rule.caretPath.replace(/\.coding\[0]$/, '');
+    const siblings = sd.rules.filter(
+      otherRule =>
+        otherRule instanceof ExportableCaretValueRule &&
+        !knownSiblings.includes(otherRule) &&
+        rule !== otherRule &&
```

Comment:
It's kind of silly, but I'd swap 162 and 163 -- as I think "don't count yourself" feels like something you'd naturally check first.  I can't explain why.  I just feel it.

I wasn't actually going to even suggest this until I got to lines 183-184 below and saw that _there_ you did put the self-check first.